modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePriceOnApplication.testcase

```diff
@@ -434,6 +434,73 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-11687. Can't checkout directly when a Price On Application Product with other normal products are in the carts, a quote must be submitted"
+	@priority = 5
+	test CanPreventPriceOnApplicationProductBeDirectlyCheckoutTogetherWithANormalProduct {
+		property portal.acceptance = "true";
+
+		task ("Given the admin enables the toggle 'Price on Application' for a product's SKU on the Test Price List 1") {
+			CommerceJSONPriceListsAndPromotionsAPI._addPriceList(
+				catalogName = "Minium",
+				currencyCode = "USD",
+				priceListName = "Test Price List 1");
+
+			CommercePriceLists.openPriceListsAdmin();
+
+			CommerceNavigator.gotoEntry(entryName = "Test Price List 1");
+
+			CommerceEntry.gotoMenuTab(menuTab = "Entries");
+
+			CommerceRelations.assignSingleRelationToProductsEntry(
+				entryName = "MIN55861",
+				title = "Add SKUs");
+
+			CommerceEntry.gotoMenuTab(menuTab = "Entries");
+
+			CommerceNavigator.gotoEntry(entryName = "MIN55861");
+
+			SelectFrame.selectFrameNoLoading(locator1 = "CommerceEntry#IFRAME_SIDE_PANEL");
+
+			CommerceSKUs.setSkuPricing(priceOnApplication = "true");
+
+			Button.clickSave();
+		}
+
+		task ("When the buyer adds two normal and one marked as 'Price on Application' products to the cart") {
+			User.logoutPG();
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer@liferay.com");
+
+			for (var productName : list "U-Joint,Mount,Torque Converters") {
+				FrontStore.addInitializerProductToCart(
+					productName = ${productName},
+					productQuantity = 1);
+			}
+		}
+
+		task ("Then the buyer can't go directly with the checkout, a quote must be submitted") {
+			CommerceNavigator.gotoMiniCart();
+
+			AssertElementPresent(
+				key_text = "Request a Quote",
+				locator1 = "Button#ANY");
+
```

Comment:
Can you please add a check to ensure that the checkout button is not able to be clicked?
Thanks!